lib/ansible/modules/files/xml.py

```diff
@@ -0,0 +1,757 @@
+            result['changed'] = True
+
+    module.exit_json(**result)
+
+
+def get_element_text(module, tree, xpath, namespaces):
+    if not is_node(tree, xpath, namespaces):
+        module.fail_json(msg="Xpath %s does not reference a node!" % xpath)
+
+    elements = []
+    for element in tree.xpath(xpath, namespaces=namespaces):
+        elements.append({element.tag: element.text})
+
+    finish(module, tree, xpath, namespaces, changed=False, msg=len(elements), hitcount=len(elements), matches=elements)
+
+
+def get_element_attr(module, tree, xpath, namespaces):
+    if not is_node(tree, xpath, namespaces):
+        module.fail_json(msg="Xpath %s does not reference a node!" % xpath)
+
+    elements = []
+    for element in tree.xpath(xpath, namespaces=namespaces):
+        child = {}
+        for key in element.keys():
+            value = element.get(key)
+            child.update({key: value})
+        elements.append({element.tag: child})
+
+    finish(module, tree, xpath, namespaces, changed=False, msg=len(elements), hitcount=len(elements), matches=elements)
+
+
+def child_to_element(module, child, in_type):
+    if in_type == 'xml':
+        infile = BytesIO(child.encode('utf-8'))
+
+        try:
+            parser = etree.XMLParser()
+            node = etree.parse(infile, parser)
+            return node.getroot()
+        except etree.XMLSyntaxError:
+            e = get_exception()
+            module.fail_json(msg="Error while parsing child element: %s" % e)
+    elif in_type == 'yaml':
+        if isinstance(child, string_types):
+            return etree.Element(child)
+        elif isinstance(child, dict):
+            if len(child) > 1:
+                module.fail_json(msg="Can only create children from hashes with one key")
+
+            (key, value) = next(iteritems(child))
+            if isinstance(value, dict):
+                children = value.pop('_', None)
+
+                node = etree.Element(key, value)
+
+                if children is not None:
+                    if not isinstance(children, list):
+                        module.fail_json(msg="Invalid children type: %s, must be list." % type(children))
+
+                    subnodes = children_to_nodes(module, children)
+                    node.extend(subnodes)
+            else:
+                node = etree.Element(key)
+                node.text = value
+            return node
+        else:
+            module.fail_json(msg="Invalid child type: %s. Children must be either strings or hashes." % type(child))
+    else:
+        module.fail_json(msg="Invalid child input type: %s. Type must be either xml or yaml." % in_type)
+
+
+def children_to_nodes(module=None, children=[], type='yaml'):
+    """turn a str/hash/list of str&hash into a list of elements"""
+    return [child_to_element(module, child, type) for child in children]
+
+
+def finish(module, tree, xpath, namespaces, changed=False, msg="", hitcount=0, matches=[]):
```

Comment:
Shouldn't use a mutable container as default arg in python.  If you ever add to the container you then find that you're sharing a single opy of that container everytime the function in python.  You can do one of these two instead:
``` python
# If you don't need to add to the container now.  This just prevents doing something stupid when the code is changed in the future.
 def finish(module, tree, xpath, namespaces, changed=False, msg="", hitcount=0, matches=tuple()): 

# if matches needs to be a mutable container.  This creates a fresh list everytime the function is called.
def finish(module, tree, xpath, namespaces, changed=False, msg="", hitcount=0, matches=None): 
    if matches is None:
        matches = []
```